tests/test_dataset_grid.py

```diff
@@ -5,49 +5,71 @@
 import reciprocalspaceship as rs
 
 
-@pytest.mark.parametrize("sample_rate", [2.5, 3, 5])
+@pytest.mark.parametrize("sample_rate", [2.0, 3.0])
 @pytest.mark.parametrize("p1", [True, False])
-def test_to_reciprocalgrid_complex(mtz_by_spacegroup, sample_rate, p1):
+@pytest.mark.parametrize("use_sf", [True, False])
+def test_to_reciprocalgrid_gemmi(mtz_by_spacegroup, sample_rate, p1, use_sf):
     """
-    Test DataSet.to_reciprocalgrid() against gemmi for complex  data
+    Test DataSet.to_reciprocalgrid() against gemmi for float data and complex
+    structure factors
     """
     if p1:
         dataset = rs.read_mtz(mtz_by_spacegroup[:-4] + "_p1.mtz")
     else:
         dataset = rs.read_mtz(mtz_by_spacegroup)
+
     gemmimtz = dataset.to_gemmi()
+    gridsize = dataset.get_gridsize(sample_rate=sample_rate)
 
-    # Note: Use P1 data to determine proper gridsize
-    testp1 = dataset.expand_to_p1().sort_index()
-    testp1.spacegroup = dataset.spacegroup
-    gridsize = testp1.to_gemmi().get_size_for_hkl(sample_rate=sample_rate)
+    if use_sf:
+        gemmigrid = gemmimtz.get_f_phi_on_grid("FMODEL", "PHIFMODEL", size=gridsize)
+        expected = np.array(gemmigrid, copy=False)
+        dataset["sf"] = dataset.to_structurefactor("FMODEL", "PHIFMODEL")
+        result = dataset.to_reciprocalgrid("sf", gridsize=gridsize)
 
-    gemmigrid = gemmimtz.get_f_phi_on_grid("FMODEL", "PHIFMODEL", size=gridsize)
-    expected = np.array(gemmigrid)
-    dataset["sf"] = dataset.to_structurefactor("FMODEL", "PHIFMODEL")
-    result = dataset.to_reciprocalgrid("sf", gridsize)
-    assert np.allclose(result, expected, rtol=1e-4)
+        # Requires rtol due to truncations applied in gemmi
+        assert np.allclose(result, expected, rtol=1e-4)
+    else:
+        gemmigrid = gemmimtz.get_value_on_grid("FMODEL", size=gridsize)
+        expected = np.array(gemmigrid, copy=False)
+        result = dataset.to_reciprocalgrid("FMODEL", gridsize=gridsize)
+        assert np.allclose(result, expected)
 
 
-@pytest.mark.parametrize("sample_rate", [2.5, 3, 5])
-@pytest.mark.parametrize("p1", [True, False])
-def test_to_reciprocalgrid_float(mtz_by_spacegroup, sample_rate, p1):
+@pytest.mark.parametrize("sample_rate", [2.0, 3.0])
+@pytest.mark.parametrize("column", ["FMODEL", "sf"])
+def test_to_reciprocalgrid_p1(mtz_by_spacegroup, sample_rate, column):
```

Comment:
Not sure this test is necessary, but I guess it doesn't hurt to have it.

Reply:
This test is very similar to the ones done for `expand_to_p1()`. Though somewhat repetitive, it seemed like a good idea to ensure that there is no dependency on `sample_rate`